src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -54,6 +54,7 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
                 syntax.Initializer,
                 receiverType,
                 syntax.Expression,
+                isNew: true,
```

Comment:
>isNew [](start = 16, length = 5)

Maybe `IsNew` isn't the greatest name here. Perhaps something like `IsNewObject` or `IsNewOrWith`? `IsNew` on it's own implies a literal `new` syntax node to me.